agenta-cli/sdk.py

```diff
@@ -0,0 +1,652 @@
+from typing import Any, Dict, Optional
+
+from pydantic import BaseModel
+
+
+class Reference(BaseModel):
+    # IDENTIFIER
+    id: Optional[str] = None
+    slug: Optional[str] = None
+    # VERSION
+    version: Optional[str] = None
+
+
+class Application(BaseModel):
+    # SCOPE
+    project_id: str
+    # IDENTIFIER
+    id: str
+    slug: str
+    # HEADER
+    name: Optional[str] = None
+    description: Optional[str] = None
+
+
+class Service(BaseModel):
+    # INTERNALS
+    url: str
+
+
+class Configuration(BaseModel):
+    # SCOPE
+    project_id: str
+    # IDENTIFIER
+    id: str
+    slug: str
+    # HEADER
+    name: Optional[str] = None
+    description: Optional[str] = None
+    # VERSION
+    version: str
+    latest: bool
+
+    # REFERENCES
+    application: Application
+
+    # INTERNALS
+    parameters: Dict[str, Any]
+
+
+class Variant(BaseModel):
+    # SCOPE
+    project_id: str
+    # IDENTIFIER
+    id: str
+    slug: str
+    # HEADER
+    name: Optional[str] = None
+    description: Optional[str] = None
+    # VERSION
+    version: str
+    latest: bool
+
+    # REFERENCES
+    application: Application
+    service: Service
+    configuration: Configuration
+
+
+class Environment(BaseModel):
+    # SCOPE
+    project_id: str
+    # IDENTIFIER
+    id: str
+    slug: str
+    # HEADER
+    name: Optional[str] = None
+    description: Optional[str] = None
+    # VERSION
+    version: str
+    latest: bool
+
+    # REFERENCES
+    application: Application
+    variant: Variant
+
+
+class Prompt(Variant):
+    pass
+
+
+class ApplicationsClient:
+    def __init__(self):
+        pass
+
+    def create(
+        self,
+        *,
+        application_slug: Optional[str] = None,
+    ) -> Prompt:
+        pass
+
+    def fetch(
+        self,
+        *,
+        configuration_reference: Optional[Reference] = None,
+        variant_reference: Optional[Reference] = None,
+        environment_reference: Optional[str] = None,
+    ) -> Prompt:
+        pass
+
+
+class ConfigurationsClient:
```

Comment:
As mentioned above, I would not break things into VariantClient, PromptClient, ConfigurationClient. It's just one client (and maybe a PromptClient as a wrapper/alias), but from the point of the API it's one route. 
I don't see the advantage of adding all the complexity at the current stage

Reply:
I added it all for completeness wrt the _Applications Structure_ RFC.
But I would only expose _PromptClient_. That's it.

Reply:
```
class PromptsClient:
    def __init__(self):
        pass

    def fork(
        self,
        *,
        parameters: Optional[Dict[str, Any]] = None,
        // AND
        app_id: Optional[str] = None,
        /// OR 
        prompt_ref: Optional[Reference] = None,
        /// OR
        prompt: Optional[Prompt] = None,
    ) -> Prompt:
        pass

    def fetch(
        self,
        *,
        prompt_ref: Optional[Reference] = None,
        // OR
        environment_ref: Optional[Reference] = None,
    ) -> Prompt:
        pass

    def commit(
        self,
        *,
        parameters: Optional[Dict[str, Any]] = None,
        prompt_ref: Optional[Reference] = None,
        // OR 
        prompt: Optional[Prompt] = None,
    ) -> Prompt:
        pass

    def deploy(
        self,
        *,
        environment_ref: Reference,
        // AND
        prompt_ref: Optional[Reference] = None,
        /// OR
        prompt: Optional[Variant] = None,
    ) -> Environment:
        pass
```